hack/dev-deploy.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+IMG=$1
+TMP_DIR="$(git rev-parse --show-toplevel)/tmp"
+
+mkdir -p ${TMP_DIR}
+
+cat << EOF | tee ${TMP_DIR}/kustomization.yaml
```

Comment:
What about moving this to `config/dev`, and move what is left of the script to the `Makefile`? We could then probably reuse the `deploy` target (and make the folder it deploys configurable), and move the `rollout status` check to the `e2e.yaml`.

Reply:
`kustomize edit set image` does modify the kustomization.yaml making the repo dirty.

Reply:
I forgot about this minor detail in the example illustrated above but I still think it does not make sense to introduce an alternative way of doing or running things (including a new `make` target) because it would make the repository dirty for CI (as this is the only place we generally care about this).

I do however not have an alternative solution on the spot, so this will do for now.

Reply:
I got rid of the script. The `make dev-deploy` is for local development and CI. After a code change you would do:
```
make test docker-build docker-push dev-deploy IMG=hiddeco/sourcer:my-test-1
```